packages/process/src/node/raw-process.ts

```diff
@@ -116,27 +116,50 @@ export class RawProcess extends Process {
                 this.emitOnError(error as ProcessErrorEvent);
             });
             this.process.on('exit', (exitCode: number, signal: string) => {
+
+            });
+
+            // When no stdio option is passed, it is null by default.
+            this.outputStream = this.process.stdout || new DevNullStream({ autoDestroy: true });
+            this.inputStream = this.process.stdin || new DevNullStream({ autoDestroy: true });
+            this.errorStream = this.process.stderr || new DevNullStream({ autoDestroy: true });
+
+            const onExit = new Promise<{ code?: number, signal?: string }>(resolve => {
+                this.process!.on('exit', (code, signal) => resolve({ code, signal }));
+            });
+            const outputClosed = new Promise(resolve => this.outputStream.on('close', () => resolve()));
+            const inputClosed = new Promise(resolve => this.inputStream.on('close', () => resolve()));
+            const errorClosed = new Promise(resolve => this.errorStream.on('close', () => resolve()));
+
+            onExit.then(exitStatus => {
+                const { code: exitCode, signal } = exitStatus;
                 // node's child_process exit sets the unused parameter to null,
                 // but we want it to be undefined instead.
                 this.emitOnExit(
-                    exitCode !== null ? exitCode : undefined,
-                    signal !== null ? signal : undefined,
+                    typeof exitCode === 'number' ? exitCode : undefined,
+                    typeof signal === 'string' ? signal : undefined,
                 );
             });
 
-            this.outputStream = this.process.stdout || new DevNullStream();
-            this.inputStream = this.process.stdin || new DevNullStream();
-            this.errorStream = this.process.stderr || new DevNullStream();
+            Promise.all([onExit, outputClosed, inputClosed, errorClosed]).then(resolved => {
```

Comment:
should not it be simple:
```ts
this.process.on('exit', (code, signal) => this.emitOnClose(code, signal));
```

Why do we need to listen for all other streams? It should be done by Node.js runtime already.

Reply:
You are right, forgot about this event.